src/test/kotlin/com/igorwojda/list/product/challenge.kt

```diff
@@ -4,7 +4,9 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun product(list: List<Int>): Int {
-    TODO("not implemented")
+    return list.reduce { product, element ->
+        product * element
```

Comment:
Very good! But, you're still missing point 3 of the challenge: A method reference (to the above operator): https://khan.github.io/kotlin-for-python-developers/#function-references